src/main/java/org/codehaus/plexus/archiver/AbstractUnArchiver.java

```diff
@@ -343,9 +343,17 @@ protected void extractFile( final File srcF, final File dir, final InputStream c
 
         try
         {
-            if ( !isOverwrite() && f.exists() && ( f.lastModified() >= entryDate.getTime() ) )
+            if ( !isOverwrite() && f.exists() )
             {
-                return;
+                final File entryFile = FileUtils.getFile( entryName );
+                if ( !StringUtils.equals( canonicalDestPath, entryFile.getAbsolutePath() ) )
```

Comment:
This check is not correct. First of all if you don't extract in the current directory it will always be `true`. For example when I ran one of the test cases `canonicalDestPath` is `C:\git\plexus-archiver\target\zip-unarchiver-tests\resources\artifactId` and `entryFile.getAbsolutePath()` is `C:\Users\Plamen\git\plexus-archiver\resources\artifactId`. Also I've made some quick tests on Windows and `canonicalDestPath` and `entryFile.getAbsolutePath()` are always with the same case. I do believe that is the case for all OSes as it is represent path not existing file (I could be wrong as the canonical path is system dependent) so it won't work even if the paths were resolved properly.

What is more the canonical and the absolute path could be different for a couple of reasons - including resolving symbolic links - just comparing the two strings is not reliable way to determine if they represent two files with the same name with different case of the letters.